source/solvers/fluid_dynamics_matrix_based.cc

```diff
@@ -1489,8 +1489,12 @@ FluidDynamicsMatrixBased<dim>::setup_AMG()
   parameter_ml.set("coarse: ifpack level-of-fill", ilu_fill);
   parameter_ml.set("coarse: ifpack absolute threshold", ilu_atol);
   parameter_ml.set("coarse: ifpack relative threshold", ilu_rtol);
-  amg_preconditioner = std::make_shared<TrilinosWrappers::PreconditionAMG>();
-  amg_preconditioner->initialize(system_matrix, parameter_ml);
+  auto new_amg_preconditioner =
+    std::make_shared<TrilinosWrappers::PreconditionAMG>();
+  new_amg_preconditioner->initialize(system_matrix, parameter_ml);
+  amg_preconditioner.swap(new_amg_preconditioner);
+  // amg_preconditioner = std::make_shared<TrilinosWrappers::PreconditionAMG>();
+  // amg_preconditioner->initialize(system_matrix, parameter_ml);
```

Comment:
I don't understand why the previous lines were removed.

Also, I would remove commented lines if possible.

Reply:
Good catch!! This was done during debugging an I forgot to get back to it (the week after ^^') I'll look into it!